modules/platforms/cpp/schema/BinaryTupleBuilder.cpp

```diff
@@ -0,0 +1,178 @@
+            throw std::logic_error("Unsupported type " + std::to_string(static_cast<int>(type)));
+    }
+}
+
+void BinaryTupleBuilder::putBytes(BytesView bytes) {
+    assert(elementIndex < elementCount);
+    assert(nextValue + bytes.size() <= valueBase + valueAreaSize);
+    std::memcpy(nextValue, bytes.data(), bytes.size());
+    nextValue += bytes.size();
+    appendEntry();
+}
+
+void BinaryTupleBuilder::putInt8(BytesView bytes) {
+    SizeT size = sizeOfInt8(BinaryTupleParser::getInt8(bytes));
+    assert(size <= bytes.size());
+    putBytes(BytesView{bytes.data(), size});
+}
+
+void BinaryTupleBuilder::putInt16(BytesView bytes) {
+    SizeT size = sizeOfInt16(BinaryTupleParser::getInt16(bytes));
+    assert(size <= bytes.size());
+    static_assert(BYTE_ORDER == LITTLE_ENDIAN);
+    putBytes(BytesView{bytes.data(), size});
+}
+
+void BinaryTupleBuilder::putInt32(BytesView bytes) {
+    SizeT size = sizeOfInt32(BinaryTupleParser::getInt32(bytes));
+    assert(size <= bytes.size());
+    static_assert(BYTE_ORDER == LITTLE_ENDIAN);
+    putBytes(BytesView{bytes.data(), size});
+}
+
+void BinaryTupleBuilder::putInt64(BytesView bytes) {
+    SizeT size = sizeOfInt64(BinaryTupleParser::getInt64(bytes));
+    assert(size <= bytes.size());
+    static_assert(BYTE_ORDER == LITTLE_ENDIAN);
+    putBytes(BytesView{bytes.data(), size});
+}
+
+void BinaryTupleBuilder::putFloat(BytesView bytes) {
+    SizeT size = sizeOfFloat(BinaryTupleParser::getFloat(bytes));
+    assert(size <= bytes.size());
+    putBytes(BytesView{bytes.data(), size});
+}
+
+void BinaryTupleBuilder::putDouble(BytesView bytes) {
+    double value = BinaryTupleParser::getDouble(bytes);
+    SizeT size = sizeOfDouble(value);
+    assert(size <= bytes.size());
+    if (size != sizeof(float)) {
+        putBytes(BytesView{bytes.data(), size});
+    } else {
+        float floatValue = value;
+        putBytes(BytesView{reinterpret_cast<std::byte*>(&floatValue), sizeof(float)});
+    }
+}
+
+void BinaryTupleBuilder::append(DATA_TYPE type, const BytesView &bytes) {
+    switch (type) {
+        case DATA_TYPE::INT8:
+            return putInt8(bytes);
+        case DATA_TYPE::INT16:
+            return putInt16(bytes);
+        case DATA_TYPE::INT32:
+            return putInt32(bytes);
+        case DATA_TYPE::INT64:
+            return putInt64(bytes);
+        case DATA_TYPE::FLOAT:
+            return putFloat(bytes);
+        case DATA_TYPE::DOUBLE:
+            return putDouble(bytes);
+        case DATA_TYPE::STRING:
+        case DATA_TYPE::BINARY:
+            return putBytes(bytes);
+
+        case DATA_TYPE::UUID:
+        case DATA_TYPE::DATE:
+        case DATA_TYPE::TIME:
+        case DATA_TYPE::DATETIME:
+        case DATA_TYPE::TIMESTAMP:
+        case DATA_TYPE::BITMASK:
+        case DATA_TYPE::NUMBER:
+        case DATA_TYPE::DECIMAL:
+        default:
+            throw std::logic_error("Unsupported type " + std::to_string(static_cast<int>(type)));
```

Comment:
Do we need a TODO here? Why are they unsupported?

Reply:
This will be addressed under IGNITE-17401 ticket.

Reply:
Let's add a TODO comment for that then.